src/lib/operators/join_sort_merge/column_materializer.hpp

```diff
@@ -70,26 +66,29 @@ class ColumnMaterializer {
     auto output = std::make_unique<MaterializedSegmentList<T>>(chunk_count);
     auto null_rows = std::make_unique<PosList>();
 
-    std::vector<T> samples;
-    samples.reserve(chunk_count * samples_per_chunk);
-    auto samples_written = ChunkOffset{0};
+    std::vector<SubSample<T>> subsamples;
+    subsamples.reserve(chunk_count);
 
     std::vector<std::shared_ptr<AbstractTask>> jobs;
     for (ChunkID chunk_id{0}; chunk_id < chunk_count; ++chunk_id) {
       const auto samples_to_write = std::min(samples_per_chunk, input->get_chunk(chunk_id)->size());
-      SubSample<T> sub_sample(samples_to_write, samples_written, samples);
+      subsamples.push_back(SubSample<T>(samples_to_write));
 
       jobs.push_back(
-          _create_chunk_materialization_job(output, null_rows, chunk_id, input, column_id, std::move(sub_sample)));
+          _create_chunk_materialization_job(output, null_rows, chunk_id, input, column_id, subsamples.back()));
       jobs.back()->schedule();
-
-      samples_written += samples_to_write;
     }
 
-    samples.resize(samples_written);
     CurrentScheduler::wait_for_tasks(jobs);
 
-    return {std::move(output), std::move(null_rows), std::move(samples)};
+    auto gathered_samples = std::vector<T>();
+    gathered_samples.reserve(samples_per_chunk * chunk_count);
+
+    for (const auto& subsample : subsamples) {
+      gathered_samples.insert(gathered_samples.end(), subsample.samples.begin(), subsample.samples.end());
```

Comment:
Not really sure how to use `move` here. :-/
```
for (auto&& subsample : subsamples) {
  ????;
}
```

Reply:
Ah, as we are working on iterators, this becomes a tad more complicated. You could do

```
gathered_samples.insert(gathered_samples.end(), std::make_move_iterator(subsample.samples.begin()), std::make_move_iterator(subsample.samples.end()));
```

Let's wait with that until it actually pops up in the profiler.

Reply:
Since we collect a maximum of 10 items per chunk, we'll probably never see this part becoming an issue. Anyhow, interesting to see that such constructs exist.